jenkinsci/tuleap-api-plugin
# Add the query to get a new access token by using the refresh token

This is a part of [story #14018](https://tuleap.net/plugins/tracker/?aid=14018) have a central management of users and groups using oauth

=> No functional change expected

In the case if you want to test functionaly:
 You can test this commit with the tuleap-oauth plugin (not released yet):
     - Pull "test/tuleap-api-integration branch" from tuleap-oauth plugin. (The master branch is not compatible with this commit yet)
     - You have to install this plugin manually :
      - In this plugin, execute: mvn clean install (or mvn package)
      - Go to the tuleap-oauth plugin directory
      - mvn install:install-file -DgroupId=io.jenkins.plugins -DartifactId=tuleap-api -Dversion=1.0.3-SNAPSHOT -Dfile=<path of your tuleap-api.hpi file> -Dpackaging=hpi
      - mvn install:install-file -DgroupId=io.jenkins.plugins -DartifactId=tuleap-api -Dversion=1.0.3-SNAPSHOT -Dfile=<path of your tuleap-api.jar file>
    - In TuleapSecurityRealm class :
      - Add offline_access scope.
      - L223 add : AccessToken refreshToken = this.accessTokenApi.getRefreshToken(accessToken,this.clientId,this.clientSecret);
      - You can also add logger to compare the first access with the new access token
      - mvn hpi:run
=> No functionnal change expected
=> If you added the logger, the access token should be different after retrieving the new access token
